applications/GeoMechanicsApplication/custom_utilities/check_utilities.h

```diff
@@ -39,4 +40,136 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) CheckUtilities
 private:
     static std::string PrintVectorContent(const std::vector<size_t>& rVector);
 }; /* Class CheckUtilities*/
+
+class KRATOS_API(GEO_MECHANICS_APPLICATION) CheckProperties
+{
+public:
+    enum class Bounds {
+        AllInclusive,
+        AllExclusive,
+        InclusiveLowerAndExclusiveUpper,
+        ExclusiveLowerAndInclusiveUpper
+    };
+
+    CheckProperties(const Properties& rProperties, const std::string& rPrintName, Bounds RangeBoundsType)
+        : mrProperties(rProperties), mrPrintName(rPrintName), mId(rProperties.Id()), mRangeBoundsType(RangeBoundsType)
+    {
+    }
+
+    CheckProperties(const Properties& rProperties, const std::string& rPrintName, std::size_t Id, Bounds RangeBoundsType)
+        : mrProperties(rProperties), mrPrintName(rPrintName), mId(Id), mRangeBoundsType(RangeBoundsType)
+    {
+    }
+
+    CheckProperties(const CheckProperties&) = delete;
+
+    CheckProperties(CheckProperties&&) = default;
+
+    CheckProperties SingleUseBounds(Bounds RangeBoundsType) const
+    {
+        return CheckProperties(mrProperties, mrPrintName, mId, RangeBoundsType);
+    }
+
+    void SetNewBounds(Bounds RangeBoundsType) const
+    {
+        mHistoryOfTypes.push_back(mRangeBoundsType);
+        mRangeBoundsType = RangeBoundsType;
+    }
+
+    void RestorePreviousBounds() const
+    {
+        if (!mHistoryOfTypes.empty()) {
+            mRangeBoundsType = mHistoryOfTypes.back();
+            mHistoryOfTypes.pop_back();
+        }
+    }
+
+    template <typename T>
+    void Check(const Variable<T>& rVariable) const
+    {
+        CheckAvailabilityOnly(rVariable);
+        CheckRangeBounds(rVariable, mDefaultLowerBound, mDefaultUpperBound);
+    }
+
+    template <typename T>
+    void Check(const Variable<T>& rVariable, double UpperBound) const
+    {
+        CheckAvailabilityOnly(rVariable);
+        CheckRangeBounds(rVariable, mDefaultLowerBound, UpperBound);
+    }
+
+    template <typename T>
+    void Check(const Variable<T>& rVariable, double LowerBound, double UpperBound) const
+    {
+        CheckAvailabilityOnly(rVariable);
+        CheckRangeBounds(rVariable, LowerBound, UpperBound);
+    }
+
+    template <typename T>
+    void CheckAvailabilityOnly(const Variable<T>& rVariable) const
+    {
+        if (!mrProperties.Has(rVariable))
+            KRATOS_ERROR << rVariable.Name() << " does not exist in the " << mrPrintName << " "
+                         << mId << "." << std::endl;
```

Comment:
I'd reword this slightly for clarity
```suggestion
            KRATOS_ERROR << rVariable.Name() << " does not exist in the " << mrPrintName << " with Id "
                         << mId << "." << std::endl;
```

Reply:
done